cookiecutter/parsers.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+
+import logging
+import yaml
+import json
+import os.path
+
+from collections import OrderedDict
+
+from .utils import work_in
+from .exceptions import ParsingError, ContextDecodingException
+
+
+DEFAULT_CONFIG_BASE = 'cookiecutter'
+
+
+def ordered_load(stream, Loader=yaml.Loader, object_pairs_hook=OrderedDict):
+    class OrderedLoader(Loader):
+        pass
+
+    def construct_mapping(loader, node):
+        loader.flatten_mapping(node)
+        return object_pairs_hook(loader.construct_pairs(node))
+    OrderedLoader.add_constructor(
+        yaml.resolver.BaseResolver.DEFAULT_MAPPING_TAG,
+        construct_mapping)
+    return yaml.load(stream, OrderedLoader)
+
+
+def load_from_yaml(fh):
+    try:
+        obj = ordered_load(fh, yaml.SafeLoader)
+    except yaml.scanner.ScannerError as e:
+        # YAML parsing error.  Let's throw a normalized exception
+        # to be catched upwards.
+        raise ParsingError(
+            'line {0}: {1}'.format(
+                e.problem_mark.line,
+                e.problem))
+    if "context" in obj and isinstance(obj["context"], dict):
+        return {"__meta__": obj["context"]}
+    return obj
+
+
+def load_from_json(fh):
+    try:
+        obj = json.load(fh, object_pairs_hook=OrderedDict)
+    except ValueError as e:
+        # JSON parsing error.  Let's throw a normalized exception
+        # to be catched upwards.
+        raise ParsingError(str(e))
+    return obj
+
+
+PARSERS = [
+    ('yml', ("YAML", load_from_yaml)),
+    ('json', ("JSON", load_from_json)),
+]
+
+
+def find_cfg_file(repo_dir):
```

Comment:
`cfg` seems a little out of place compared to the rest of the functions, I'd prefer the slightly more verbose `config`?